core/runtime/common/stack_limiter.cpp

```diff
@@ -0,0 +1,848 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "runtime/common/stack_limiter.hpp"
+
+#include "common/visitor.hpp"
+#include "log/logger.hpp"
+#include "log/profiling_logger.hpp"
+
+#include <wabt/binary-reader-ir.h>
+#include <wabt/binary-reader.h>
+#include <wabt/binary-writer.h>
+#include <wabt/ir.h>
+#include <wabt/validator.h>
+
+namespace kagome::runtime {
+
+  class Stream final : public wabt::Stream {
+   public:
+   protected:
+    wabt::Result WriteDataImpl(size_t offset,
+                               const void *data,
+                               size_t size) override {
+      if (offset + size > bytes.size()) {
+        bytes.resize(offset + size);
+      }
+      std::copy_n(reinterpret_cast<const uint8_t *>(data),
+                  size,
+                  bytes.begin() + offset);
```

Comment:
`memcpy`

Reply:
No? Why?

Reply:
Because that's bytes (`std::copy` is not optimized and slow).

`memmove` is required (correct overlap handling, implied by `Move` in `MoveDataImpl`)